src/main/java/pl/betoncraft/betonquest/conditions/TestForBlockCondition.java

```diff
@@ -34,18 +35,22 @@
 
 	private final LocationData loc;
 	private final Material material;
+	private final byte data;
 
 	public TestForBlockCondition(Instruction instruction) throws InstructionParseException {
 		super(instruction);
 		staticness = true;
 		persistent = true;
 		loc = instruction.getLocation();
 		material = instruction.getEnum(Material.class);
+		data = instruction.getByte(instruction.getOptional("data"), (byte) 0);
 	}
 
 	@Override
+	@SuppressWarnings("deprecation")
 	public boolean check(String playerID) throws QuestRuntimeException {
-		return loc.getLocation(playerID).getBlock().getType().equals(material);
+		Block block = loc.getLocation(playerID).getBlock();
+		return (block.getType() == material && block.getData() == data);
```

Comment:
It doesn't match the previous behavior. If you don't specify any `data` the condition will specifically require it to be 0, whereas before it allowed any value.

I suggest wrapping that `byte` in `Optional<Byte>` and doing the equality check only if it's present. If course any other working solution will be good too :smile: